src/test/java/org/jenkinsci/remoting/engine/HandlerLoopbackLoadStress.java

```diff
@@ -0,0 +1,782 @@
+
+    private final ServerSocketChannel serverSocketChannel;
+
+    private final Acceptor acceptor;
+
+    private final KeyPair keyPair;
+    private final X509Certificate certificate;
+
+    private final JnlpProtocolHandler<? extends JnlpConnectionState> handler;
+
+    private final SettableFuture<SocketAddress> addr = SettableFuture.create();
+    private final Random entropy = new Random();
+
+    private final RuntimeMXBean runtimeMXBean;
+    private final OperatingSystemMXBean operatingSystemMXBean;
+    private final Method _getProcessCpuTime;
+    private final Config config;
+    private final Stats stats;
+
+    public HandlerLoopbackLoadStress(Config config)
+            throws IOException, NoSuchAlgorithmException, CertificateException, KeyStoreException,
+            UnrecoverableKeyException, KeyManagementException, OperatorCreationException {
+        this.config = config;
+        KeyPairGenerator gen = KeyPairGenerator.getInstance("RSA");
+        gen.initialize(2048); // maximum supported by JVM with export restrictions
+        keyPair = gen.generateKeyPair();
+
+        Date now = new Date();
+        Date firstDate = new Date(now.getTime() + TimeUnit.DAYS.toMillis(10));
+        Date lastDate = new Date(now.getTime() + TimeUnit.DAYS.toMillis(-10));
+
+        SubjectPublicKeyInfo subjectPublicKeyInfo =
+                SubjectPublicKeyInfo.getInstance(keyPair.getPublic().getEncoded());
+
+        X500NameBuilder nameBuilder = new X500NameBuilder(BCStyle.INSTANCE);
+        X500Name subject = nameBuilder
+                .addRDN(BCStyle.CN, getClass().getSimpleName())
+                .addRDN(BCStyle.C, "US")
+                .build();
+
+        X509v3CertificateBuilder certGen = new X509v3CertificateBuilder(
+                subject,
+                BigInteger.ONE,
+                firstDate,
+                lastDate,
+                subject,
+                subjectPublicKeyInfo
+        );
+
+        JcaX509ExtensionUtils instance = new JcaX509ExtensionUtils();
+
+        certGen.addExtension(X509Extension.subjectKeyIdentifier,
+                false,
+                instance.createSubjectKeyIdentifier(subjectPublicKeyInfo)
+        );
+
+        ContentSigner signer = new JcaContentSignerBuilder("SHA1withRSA")
+                .setProvider(BOUNCY_CASTLE_PROVIDER)
+                .build(keyPair.getPrivate());
+
+        certificate = new JcaX509CertificateConverter()
+                .setProvider(BOUNCY_CASTLE_PROVIDER)
+                .getCertificate(certGen.build(signer));
+
+        char[] password = "password".toCharArray();
+
+        KeyStore store = KeyStore.getInstance("jks");
+        store.load(null, password);
+        store.setKeyEntry("alias", keyPair.getPrivate(), password, new Certificate[]{certificate});
+
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        kmf.init(store, password);
+
+        context = SSLContext.getInstance("TLS");
```

Comment:
From Oracle's Java SE 7 Javadoc: "Every implementation of the Java platform is required to support the following standard SSLContext protocol: TLSv1". Nothing is said for SE6, which needs to be supported by remoting right now. Just "TLS" smells like a bug in any case